java/ql/lib/semmle/code/xml/AndroidManifest.qll

```diff
@@ -51,6 +61,23 @@ class AndroidApplicationXmlElement extends XMLElement {
    * Gets a component child element of this `<application>` element.
    */
   AndroidComponentXmlElement getAComponentElement() { result = this.getAChild() }
+
+  /**
+   * Holds if this application element has the attribute `android:debuggable` set to `true`.
+   */
+  predicate isDebuggable() {
+    exists(AndroidXmlAttribute attr |
+      this.getAnAttribute() = attr and
+      attr.getName() = "debuggable" and
+      attr.getValue() = "true"
+    )
+  }
+
+  /**
+   * Overrides the getFile() predicate of the XMLElement class to get the
+   * AndroidManifest.xml file itself.
+   */
+  override AndroidManifestXmlFile getFile() { result = super.getFile() }
```

Comment:
This override shouldn't be necessary. If you did this to be able to access `AndroidManifestXmlFile` predicates, see my comment below about casting:

```suggestion
```

Reply:
Removed in commit [fe127e5](https://github.com/github/codeql/pull/9939/commits/fe127e5adccc58a83b6c6268212d2a1bba7f4377).